src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private Node<T> left;
+        private Node<T> right;
+        private T value;
+
+        Node(Node<T> left, T value, Node<T> right) {
+            this.value = value;
+            this.left = left;
+            this.right = right;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> addedElement = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = addedElement;
+            last = addedElement;
+        } else {
+            addedElement.left = last;
+            last.right = addedElement;
+            last = addedElement;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size() || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of List bounds");
+        }
+        if (index == size()) {
+            add(value);
+            return;
+        }
+        Node<T> nextElement = first;
+        int i = 0;
+        while (i < index) {
+            nextElement = nextElement.right;
+            i++;
+        }
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html): Don’t iterate your nodes from the head to tail or vice versa.
Lets create separate private method to find element by index (we have this logic in many methods)